database/src/main/java/dao/Dao.java

```diff
@@ -0,0 +1,8 @@
+package dao;
+
+import javax.persistence.EntityManager;
+
+public interface Dao {
+    EntityManager getEntityManager();
+    EntityManager setEntityManager();
```

Comment:
if method called with adv set please transfer to the method param
setEntityManager(EntityManaget) { this.entityManager = entityManager;}